File: src/extras-kiconthemes/libkpixmapsequenceloader.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");

/// https://api.kde.org/kpixmapsequenceloader.html
pub const kpixmapsequenceloader = struct {
    /// [Qt documentation](https://api.kde.org/kpixmapsequenceloader.html#load)
    ///
    /// ``` param1: []const u8, param2: i32 ```
    pub fn Load(param1: []const u8, param2: i32) QtC.KPixmapSequence {
        const param1_str = qtc.libqt_string{
            .len = param1.len,
            .data = param1.ptr,
        };
        return qtc.KPixmapSequenceLoader_Load(param1_str, @intCast(param2));
    }
};

File: src_original/extras-kiconthemes/libkpixmapsequenceloader.cpp
#include <KPixmapSequence>
#include <KPixmapSequenceLoader>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <kpixmapsequenceloader.h>
#include "libkpixmapsequenceloader.h"
#include "libkpixmapsequenceloader.hxx"

KPixmapSequence* KPixmapSequenceLoader_Load(const libqt_string param1, int param2) {
    QString param1_QString = QString::fromUtf8(param1.data, param1.len);
    return new KPixmapSequence(KPixmapSequenceLoader::load(param1_QString, static_cast<int>(param2)));
}
